### ISSUE TITLE
Configured region not honored with de-informal language

### ISSUE THREAD

**Issue by stapelberg** [pretix/pretix#3947]
### Problem and impact

When setting my user account’s language to de-informal, the configured region (CH) is no longer honored for formatting money values, e.g. in the orders page.

### Steps to reproduce

1. Configure region to CH
2. Configure language to de; orders page lists “CHF 97.56” (with dots)
3. Configure language to de-informal; orders page lists “97,56 CHF” (with commas)

Looking at the code, I think the problem is that the language function treats “informal” as a region: https://github.com/pretix/pretix/blob/6bf23b0fddbac3aa8df6f81db35c91fdcc058aba/src/pretix/base/i18n.py#L146-L154

### Version

2024.2

**Comment by raphaelm** [comment]
I'll look into it but it might be not fixable the way the informal variant is implemented, it's basically a fake region in itself :/

**Comment by stapelberg** [comment]
Wouldn’t just removing “-informal” before trying to determine the region fix the problem?

**Comment by raphaelm** [comment]
No, the region will not be added to the language in the first place if de-informal is used since "informal" is implemented like a region (think de-de, de-at, de-ch and de-informal)

**Comment by wiffbi** [comment]
The same problem arrises if one uses es-419 – Spanish for Latin America. The region is not honored and for e.g. Mexico the number format is wrong (Z#23220951). Not sure, how to fix this though. We might add a language es-mx to the supported languages, that fixes the number format, but AFAICT it will fall back to es instead of es-419.

**Comment by raphaelm** [comment]
Fix in https://github.com/pretix/pretix/pull/5814
